code/script.js

```diff
@@ -4,16 +4,117 @@
 // Your code goes here
 alert(
   `Welcome to our Javascript Pizzeria. Ready to Start? - Click 'OK' to begin.`
-)
+);
+// Ask the user for their name and store the input in a variable
+const userName = prompt(`Please enter your name:`);
 
+// Display a greeting using the user's name
+if (userName) {
+  alert(`Hello, ${userName} ! Welcome to our Javascript Pizzeria. 🍕`);
+} else {
+  alert(`No name entered.☹️`);
+};
 // Step 2 - Food choice
-// Your code goes here
+const foodChoice = prompt(`What food would you like to order? 
+ Enter a number:
+ 1 - Pizza
+ 2 - Pasta
+ 3 - Salad`);
+
+let foodType;
+
+if (foodChoice === `1`) {
+  foodType = `Pizza`;
+  alert(`You chosen  pizza.`);
+} else if (foodChoice === `2`) {
+  foodType = `Pasta`;
+  alert(`You chosen  pasta.`);
+} else if (foodChoice === `3`) {
+  foodType = `Salad`;
+  alert(`You chosen salad.`);
+} else {
+  alert(`Please select again.`);
+};
+
 
 // Step 3 - Subtype choice
 // Your code goes here
 
+let pizzaType;
+let pastaType;
+let saladType;
+
+if (foodType === `Pizza`) {
+  pizzaType= prompt(`Choose a type of Pizza
+  1 - Margherita
+  2 - Pepperoni
+  3 - Vegetarian `);
+
+  if (pizzaType === `1`) {
+        alert(`You chosen Margherita pizza.`);
+      } else if (pizzaType === `2`) {
+        alert(`You chosen Pepperoni pizza.`);
+      } else if (pizzaType === `3`) {
+        alert(`You chosen Vegetarian pizza.`);
+      } else {
+        alert(`Invalid pizza type selection.`);
+      }
+} else if (foodType === `Pasta`) {
+  pastaType = prompt(`Choose a type of Pasta
+  1 - Spaghetti
+  2 - Alfredo
+  3 - Penne`);
+  if (pastaType === `1`) {
+        alert(`You chosen Spaghetti.`);
+      } else if (pastaType === `2`) {
+        alert(`You chosen Alfredo.`);
+      } else if (pastaType === `3`) {
+        alert(`You chosen Penne .`);
+      } else {
+        alert(`Invalid pasta type selection.`);
+      }
+} else if (foodType === `Salad`) {
+  saladType = prompt(`CHoose a type of Salad
+  1 - Caesar
+  2 - Greek
+  3 - Garden`);
+  if (saladType === `1`) {
+        alert(`You chosen Caesar Salad.`);
+      } else if (saladType === `2`) {
+        alert(`You chosen Greek Salad.`);
+      } else if (saladType === `3`) {
+        alert(`You chosen Garden Salad.`);
+      } else {
+        alert(`Invalid salad type selection.`);
+      }
+} else {
+  alert(`Something went wrong. Please try again.`);
+};
+
```

Comment:
If you write, for example, 4 instead of 1-3, you get an error message, but it seems you are still sent on to the next question.